src/dune_rpc_impl/run.ml

```diff
@@ -54,9 +56,15 @@ end = struct
       let socket =
         let dir =
           Path.of_string
-            (match Xdg.runtime_dir with
-            | Some p -> p
-            | None -> Filename.get_temp_dir_name ())
+            (match Xdg.runtime_dir (Lazy.force Dune_util.xdg) with
+            | None -> Filename.get_temp_dir_name ()
+            | Some p ->
+              let p = Filename.concat p "dune/s" in
```

Comment:
I'm not super familiar with `Filename`, but should this be a nested `concat` (instead of `dune/s`?)

Reply:
Either one should work since windows will accept `/` as well.